ui/components/src/EChart.tsx

```diff
@@ -121,62 +102,80 @@ export type OnEventsType<T> = {
 export interface EChartsProps<T> {
   option: EChartsCoreOption;
   sx?: SxProps<Theme>;
-  onChartInitialized?: (instance: ECharts) => void;
   onEvents?: OnEventsType<T>;
   _instance?: React.MutableRefObject<ECharts | undefined>;
-  parentContainer?: HTMLDivElement | null;
+  onChartInitialized?: (instance: ECharts) => void;
 }
 
-export function ECharts<T>({ sx, _instance, onChartInitialized, option, onEvents }: EChartsProps<T>) {
-  const [containerRef, setContainerRef] = useState<HTMLDivElement | null>(null);
-  const [echart, setChart] = useState<ECharts | undefined>(undefined);
-  // Create a chart instance in the container
-  useLayoutEffect(() => {
-    if (containerRef === null) return;
-
-    const chart = init(containerRef);
-    setChart(chart);
-    if (_instance !== undefined) {
-      _instance.current = chart;
+export const EChart = React.memo(function EChart<T>({
```

Comment:
Renamed to avoid confusion / needing an alias to fix collision with _ECharts from 'echarts/core'_ 
Also since this component is for rendering a single chart